### DISCUSSION TITLE
Eset Nod32 Anti-Virus blocks some operation(s) within OpenShell's installer

### DISCUSSION THREAD

**Discussion by LinuxOnTheDesktop** [Open-Shell/Open-Shell-Menu#2042] [General]
Just now I installed apre-release version (version 4.4.194) of OpenShell. My anti-virus, namely Eset's 'Nod32 Anti-Virus', blocked some activity within the installer; that blocking seemed to prompt the installer to tell me that an operation had failed but could be retried.

Solutions (or at least workarounds)

1. Whilst installing OpenShell, tell Nod to pause its protection.
2. Use Nod's 'performance' whitelist to add (i.e. exclude from real-time detection) the directory that contains the installer.

I did _1_ whilst installing OpenShell (and that worked) and, post-installation, _2_.

I suppose the messing with the start button is bound to look suspicious.

My system: Windows 10 Pro - 22H2 (2009: 19045).

I put this here, as a 'discussion', because I was unsure where else to put it.

**Comment by ge0rdi** [comment]
> My anti-virus, namely Eset's 'Nod32 Anti-Virus', blocked some activity within the installer;

Could you post some logs/screenshots about what was blocked?

> that blocking seemed to prompt the installer to tell me that an operation had failed but could be retried.

What operation failed?
Do you have some screenshot?

**Comment by LinuxOnTheDesktop** [comment]
I am afraid that all I have is the following, from Eset's log.

```05/12/2024 10:17:25;C:\Windows\System32\msiexec.exe;Start new application;C:\WINDOWS\Installer\MSI5B50.tmp;Blocked;Block execution of known bad files according to LiveGrid;;6A795B8A432514F20FF884F1A5286EA82E32E150;126E1695382F60B99EA5F8BAB35873F690AE9DEA```

But perhaps that log entry pertains to something other than OpenShell. Sorry.

**Reply by ge0rdi** [reply]
Thank you.
I was able to replicate it as well.

The file in question is `SetupHelper.exe` used by our setup.
It is blocked by ESET's cloud by mistake.
I have reported it already.

Hopefully they will fix it soon.

**Comment by Anselm970** [comment]
Emsisoft Anti-Malware Home reports:

05.12.2024 21:28:40
Malware "Gen:Variant.Ser.Lazy.9240 (B)" in "D:\Install\System\Open Shell Menu\OpenShellSetup_4_4_194.exe" in Quarantäne verschoben von Hauptkomponente

05.12.2024 21:28:36
Dateiwächter-Fund: Malware "Gen:Variant.Ser.Lazy.9240 (B)" in D:\Install\System\Open Shell Menu\OpenShellSetup_4_4_194.exe (SHA1: 289C94F12CF234F86DAB949D5EDA6BA5EB6B7FEC)

05.12.2024 21:28:36
Von Regel unter Quarantäne

**Reply by Anselm970** [reply]
(English see below.)

Meldung heute früh:

Emsisoft Anti-Malware Home
Nachdem die in Quarantäne befindlichen Objekte mit den neu heruntergeladenen Signaturen erneut gescannt wurden, hat sich herausgestellt dass einige von ihnen fälschlicherweise erkannt wurden (Fehlalarme). Möchten Sie diese Objekte jetzt an ihrer Ursprungsposition wiederherstellen?
D:\Install\System\Open Shell Menu\OpenShellSetup_4_4_194.exe

Message this morning:

Emsisoft Anti-Malware Home
After rescanning the quarantined objects with the newly downloaded signatures, it has become apparent that some of them were detected incorrectly (false positives). Would you now like to restore these objects to their original position?
D:\Install\System\Open Shell Menu\OpenShellSetup_4_4_194.exe
